oprf/oprf_test.go

```diff
@@ -0,0 +1,387 @@
+		}
+	}
+}
+
+func (v *Vectors) runP384(t *testing.T) {
+	for _, j := range v.Vector {
+		srv, err := NewServer(OPRFP384, nil, nil)
+		if err != nil {
+			t.Fatal("invalid setup of server: " + err.Error())
+		}
+		if srv == nil {
+			t.Fatal("invalid setup of server: no server.")
+		}
+
+		privKey, _ := hex.DecodeString(v.PrivK)
+		srv.Kp.PrivK.Set(privKey)
+
+		client, err := NewClient(OPRFP384)
+		if err != nil {
+			t.Fatal("invalid setup of client: " + err.Error())
+		}
+
+		token, bToken := blindTest(client.suite, j)
+		testBToken, _ := hex.DecodeString(j.Blind.Blinded)
+		if !bytes.Equal(testBToken[:], bToken[:]) {
+			test.ReportError(t, bToken[:], testBToken[:], "request")
+		}
+
+		eval, _ := srv.Evaluate(bToken)
+		if eval == nil {
+			t.Fatal("invalid evaluation of server: no evaluation.")
+		}
+
+		testEval, _ := hex.DecodeString(j.Evaluation.Eval)
+		if !bytes.Equal(testEval[:], eval.element[:]) {
+			test.ReportError(t, eval.element[:], testEval[:], "eval")
+		}
+
+		info := []byte("test information")
+		iToken, h, _ := client.Finalize(token, eval, info)
+
+		testIToken, _ := hex.DecodeString(j.Unblind.IToken)
+		if !bytes.Equal(testIToken[:], iToken[:]) {
+			test.ReportError(t, iToken[:], testIToken[:], "finalize")
+		}
+
+		testOutput, _ := hex.DecodeString(j.Output)
+		if !bytes.Equal(testOutput[:], h[:]) {
+			test.ReportError(t, h[:], testOutput[:], "finalize")
+		}
+	}
+}
+
+func (v *Vectors) runP521(t *testing.T) {
+	for _, j := range v.Vector {
+		srv, err := NewServer(OPRFP521, nil, nil)
+		if err != nil {
+			t.Fatal("invalid setup of server: " + err.Error())
+		}
+		if srv == nil {
+			t.Fatal("invalid setup of server: no server.")
+		}
+
+		privKey, _ := hex.DecodeString(v.PrivK)
+		srv.Kp.PrivK.Set(privKey)
+
+		client, err := NewClient(OPRFP521)
+		if err != nil {
+			t.Fatal("invalid setup of client: " + err.Error())
+		}
+
+		token, bToken := blindTest(client.suite, j)
+		testBToken, _ := hex.DecodeString(j.Blind.Blinded)
+		if !bytes.Equal(testBToken[:], bToken[:]) {
+			test.ReportError(t, bToken[:], testBToken[:], "request")
+		}
+
+		eval, _ := srv.Evaluate(bToken)
+		if eval == nil {
+			t.Fatal("invalid evaluation of server: no evaluation.")
+		}
+
+		testEval, _ := hex.DecodeString(j.Evaluation.Eval)
+
+		if !bytes.Equal(testEval[:], eval.element[:]) {
+			test.ReportError(t, eval.element[:], testEval[:], "eval")
+		}
+
+		info := []byte("test information")
+		iToken, h, _ := client.Finalize(token, eval, info)
+
+		testIToken, _ := hex.DecodeString(j.Unblind.IToken)
+		if !bytes.Equal(testIToken[:], iToken[:]) {
+			test.ReportError(t, iToken[:], testIToken[:], "finalize")
+		}
+
+		testOutput, _ := hex.DecodeString(j.Output)
+		if !bytes.Equal(testOutput[:], h[:]) {
+			test.ReportError(t, h[:], testOutput[:], "finalize")
+		}
+	}
+}
+
+func TestDraftVectors(t *testing.T) {
+	// Test vectors from draft-05
+	var v Vectors
+
+	v.readFile(t, "testdata/256_vectors.json")
```

Comment:
How did we create these test vectors? Why are they separate for each suite? Shouldn't all vectors belong to a single JSON file?

Reply:
They are taken from the sage poc, with modifications. I thought it will be clearer to have them on separate files.

Reply:
What modifications? Ideally, the output of Sage is read directly by the test code here, with *no* modifications, so we reduce the chance of errors.

Reply:
Sage outputs the parameters with a `0x`, which golang `hex.DecodeString` cannot parse, as it does not expect the '0x'.

Reply:
So let's update Sage to drop that prefix! Simple fix. :)

Reply:
Sure.

Reply:
I will go with the opposite, make Go skip the 0x prefix on the call to `hex.DecodeString`

Reply:
ok.. I'll change them here as well.